File: tests/adapter-fixtures/typescript/calls.ts
// Fixture for call-resolution tests.
// The parser must emit simple-name Calls edges, not dotted "obj.method" strings.

function plain(): void {
    bare();
}

function member(): void {
    console.log("hi");
}

function chained(): void {
    a.b().c();
}

function optional(): void {
    obj?.maybe();
}

function constructed(): Widget {
    return new Widget();
}

class Greeter {
    greet(): void {
        this.sayHi();
        helperCall();
    }
}

// A property-chain receiver: the express hand-off shape. `this.router` is a
// property whose value comes from outside this tree, so the callee stays the
// bare leaf and the receiver is what says where it was read.
class Application {
    handle(req: Request, res: Response): void {
        this.router.handle(req, res);
        this.own();
        deps[0].run();
        make().run();
    }

    own(): void {}
}

File: tests/adapter-fixtures/go/calls.go
// Fixture for Go call-resolution tests.
// Covers plain calls, package-qualified calls, receiver-method calls,
// and chained method calls. The parser must emit simple-name Calls
// edges (e.g. "Println" rather than "fmt.Println").
package calls

import "fmt"

type Server struct{}

func (s *Server) Run() {}

func (s *Server) B() *Server { return s }

func (s *Server) C() {}

func plain() {}

func Caller() {
	plain()
	fmt.Println("x")
	s := &Server{}
	s.Run()
	s.B().C()
}
